### ISSUE TITLE
Unitize test suite; avoid unnecessary calls to DB

### ISSUE THREAD

**Issue by Datamance** [lookit/lookit-api#564]
**TL;DR**: Create proper distinction between unit and integration tests - don't pull from the DB any more than is necessary.

**Narrative**
As a developer, I want my tests to run as fast as possible (while remaining thorough). This facilitates a faster development cycle by saving precious minutes during the testing phase. If our tests are properly unitized (mocking out service dependencies), we can avoid the time-intensive churn of calling to the test database.

**Acceptance Criteria**
- [ ] All test files refactored
- [ ] All tests pass

**Implementation Notes**
One of the most notable antipatterns in our test suite is the "do-it-all" `setUp` function. As `django.test.TestCase` guarantees a clean database (running each test in a transaction and flushing afterward), `setUp` is run for literally each and every single test. While this guarantees determinism and frees us from having to worry about order of execution for individual test methods, it also means that we are generally using `setUp` incorrectly, loading all context required to run _any_ of the test methods _each time_. This is clearly inefficient and will require some potentially tedious refactoring.

Test DB initialization will also be greatly helped by the Single App consolidation.
